### ISSUE TITLE
App crashes when switching between two objects, if a feature not in the second object is selected 

### ISSUE THREAD

**Issue by wish1832** [amc-heme/scExploreR#115]
When a feature is selected in one object and does not exist in the other object, the app crashes when switching objects. 

This is likely related to the module structure for the backend of the app: a separate module exists for each object created, but the object updates reactively; the result is that the new object is being applied to the module corresponding to the previous object.

A long-term fix will involve correcting the module structure, but this will take a while. A short-term fix would be to clear the selected features when switching between objects.

**Comment by wish1832** [comment]
@jairav @srg0502 @agillen Let's check if this still occurs.

**Comment by agillen** [comment]
This appears to be limited specifically to Ridge Plots in the current main branch. No other plot types are affected. @wish1832 - any particular way that the ridge plots are handled differently that you're aware of?

**Comment by agillen** [comment]
Full error from an example multi-object browser:

```
Warning: The following keyed vars could not be found in object ‘RNA’:rna_A1BG
Attempting to pull from other locations
Warning: Error in FetchData: None of the requested variables were found: rna_A1BG
  96: <Anonymous>
  95: signalCondition
  94: signal_abort
  93: abort
  92: FetchData.Seurat
  90: expr_plot
  89: SCUBA::plot_ridge
  88: shiny_ridge
  87: <reactive>
  71: plot
  64: <reactive>
  48: current_xlim
  45: observe
  44: <observer>
   1: runApp
Warning: Error in FetchData: None of the requested variables were found: rna_A1BG
  171: <Anonymous>
  170: stop
  169: plot
  168: renderPlot
  166: func
  126: drawPlot
  112: <reactive:plotObj>
   96: drawReactive
   83: renderFunc
   82: output$flex_aml_plots-ridge-plot
    1: runApp
```

`flex_aml` is, confusingly, the _original_ object being switched _from_. This object does contain the feature of interest (`rna_A1BG`).

**Comment by wish1832** [comment]
This may be solved by putting a `req()` statement in the reactive producing the plot that tests if the features exist in the object before plotting. The `req()` statement stops plot generation until the features input is updated again, when the user enters a feature for the new object. It is not likely the user will notice anything if this is added.

**Comment by wish1832** [comment]
I have found similar errors with other plot types. When switching between SCE and anndata objects with a DimPlot enabled, I got the error below. "UMAP_1" and "UMAP_2" are the reduction coordinates corresponding to the SCE UMAP reduction, and the `py_call_impl` part of the error implies that the function is looking for these in the anndata object. This error does not crash the app, however, and the DimPlot renders correctly within a second after switching.

```
Warning: Error in py_call_impl: ValueError: None of the requested variables were found : UMAP_1, UMAP_2
Run `reticulate::py_last_error()` for details.
  197: <Anonymous>
  196: stop
  195: py_call_impl
  194: py$fetch_anndata
  193: FetchData.AnnDataR6 [/Users/billshowers/repos/SCUBA_amcheme/R/FetchData.R#465]
  191: fetch_reduction.AnnDataR6 [/Users/billshowers/repos/SCUBA_amcheme/R/fetch_reduction.R#111]
  189: SCUBA::plot_reduction [/Users/billshowers/repos/SCUBA_amcheme/R/plot_reduction.R#139]
  188: shiny_umap [/Users/billshowers/repos/scExploreR_amcheme/R/shiny_umap.R#119]
  187: <reactive> [/Users/billshowers/repos/scExploreR_amcheme/R/module-plot_module.R#3584]
  185: .func
  182: contextFunc
  181: env$runWith
  174: ctx$run
  173: self$.updateValue
  171: plot
  170: renderPlot [/Users/billshowers/repos/scExploreR_amcheme/R/module-plot_module.R#4093]
  168: func
  128: drawPlot
  114: <reactive:plotObj>
   98: drawReactive
   85: renderFunc
   84: output$SCE object_plots-dimplot-plot
    3: runApp
    2: print.shiny.appobj
    1: <Anonymous>
```

**Comment by wish1832** [comment]
> This may be solved by putting a `req()` statement in the reactive producing the plot that tests if the features exist in the object before plotting. The `req()` statement stops plot generation until the features input is updated again, when the user enters a feature for the new object. It is not likely the user will notice anything if this is added.

This approach (checking if a feature is in an object) would be a useful helper function for SCUBA: amc-heme/SCUBA#100

**Comment by wish1832** [comment]
I found a difference between ridge plots and other plots. Ridge plots depend on a reactive called `features_entered()`, while other plots based on features depend on `features()`.

**Comment by agillen** [comment]
Closed with #331.
